chore: initial TransferGo CaseStudy

Ferror opened the pull request:
Initial step for TransferGo case-study. Looking forward for suggestions and possible next steps 😄 

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Added a TransferGo case study with a public AsyncAPI remittance event-bus spec, AWS SNS/SQS channels, and a sample remittance message (RemittanceTransactionRequested).
  * Introduces a comprehensive event catalog and published workflow for consuming and publishing remittance events.

* **Documentation**
  * Adds challenge/solution narratives, technical posture (architecture, CI/CD, contract testing, tooling), detailed remittance contract (sender/receiver, amount, currency, paymentMethod) and example payload.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Ferror committed:
Initial TransferGo CaseStudy

Ferror committed:
requested changes

derberg reviewed (commented):
@Ferror thanks a lot for the PR

you need to followup and provide full example as a file under `resources/casestudies/transfergo/asyncapi.yaml`

Ferror committed:
asyncapi file

Ferror committed:
Update public/resources/casestudies/transfergo/asyncapi.yaml

Co-authored-by: coderabbitai[bot] <136622811+coderabbitai[bot]@users.noreply.github.com>

derberg commented:
looking great https://deploy-preview-4409--asyncapi-website.netlify.app/casestudies/transfergo ❤️ 

I left you some feedback in DM. One more thing, can you please add links to tools, especially open source, that you mention in the PR? when published on the website, especially so popular like ours, it is good for the open source projects to get better SEO results. You know, links to their websites, if they exists of course, like the one for Microcks and Event Catalog, and if it doesn't, then GH link

Ferror committed:
Update transfergo.yml

Ferror committed:
Update transfergo.yml

Ferror committed:
modelina

derberg commented on config/casestudies/transfergo.yml:
```diff
@@ -0,0 +1,78 @@
+  tools: |
+    TransferGo relies on [AsyncAPI CLI](https://github.com/asyncapi/cli), [AsyncAPI React](https://github.com/asyncapi/asyncapi-react), [AsyncAPI Modelina](https://github.com/asyncapi/modelina), [Port.io Developer Portal](https://www.port.io/), [Event Catalog](https://www.eventcatalog.dev/), [Microcks](https://microcks.io/), [LocalStack](https://github.com/localstack/localstack), and API Guardian ...
+  fullExample: resources/casestudies/transfergo/asyncapi.yaml
+additionalResources: ''
```
```suggestion
additionalResources: See [slides from 2023](https://drive.google.com/file/d/1t7tYMr8FMRInaZV0lWod1QfDuhhSjwBJ/view?usp=sharing) that show early days of adoption
```

Ferror committed:
Update config/casestudies/transfergo.yml

Co-authored-by: Lukasz Gornicki <lpgornicki@gmail.com>

Ferror committed:
asyncapi 3.0

derberg committed:
added more links

Added links to Event Catalog and Microcks in the documentation for better accessibility.

derberg reviewed (approved):
well done, can't wait to see the blog post

derberg committed:
Merge branch 'master' into transfergo-case-study

derberg commented:
/rtm
